pandas/core/apply.py

```diff
@@ -90,16 +90,20 @@ def frame_apply(
     kwargs=None,
 ) -> FrameApply:
     """construct and return a row or column based frame apply object"""
+    _, func, columns, _ = reconstruct_func(func, **kwargs)
+
     axis = obj._get_axis_number(axis)
     klass: type[FrameApply]
     if axis == 0:
         klass = FrameRowApply
     elif axis == 1:
+        if columns:
+            raise NotImplementedError(
+                "func given to frame_apply cannot contain "
+                "an index relabeling when axis is 1"
+            )
         klass = FrameColumnApply
 
-    _, func, _, _ = reconstruct_func(func, **kwargs)
-    assert func is not None
```

Comment:
`reconstruct_func` runs `assert func is not None` before `return` so it is redundant to do it again here